### PR TITLE
[4.0] Render label

### PR DESCRIPTION
**Pull request by brianteeman**
There is an option "hiddenLabel" for a form field. But instead of "hiding" the label it doesn't render the label at all. This causes accessibility issues because every form field input must have an associated label.

This PR changes it so that instead of the label not being rendered at all it is rendered with a class of sr-only instead.

### Testing
Add the following field to any form

```
		<field
			name="test"
			type="text"
			label="Hidden"
			hiddenLabel="true"
		/>
```

Before and After applying the PR there will be no visual change BUT if you view the source you will see that the code has changed

### Before
```
<div class="control-group">
	<div class="controls">
		<input type="text" name="jform[test]" id="jform_test" value="" class="form-control">
	</div>
</div>
```

### After

```
<div class="control-group">
	<div class="control-label sr-only">
	        <label id="jform_test-lbl" for="jform_test">Hidden</label>
	</div>
	<div class="controls has-success">
		<input type="text" name="jform[test]" id="jform_test" value="" class="form-control valid form-control-success" aria-invalid="false">
	</div>
</div>
```

### CODE REVIEW

File: layouts/joomla/form/renderfield.php [hunk 1]
```diff
@@ -30,12 +30,11 @@
 $class = empty($options['class']) ? '' : ' ' . $options['class'];
 $rel   = empty($options['rel']) ? '' : ' ' . $options['rel'];
 $id    = $name . '-desc';
+$hide  = empty($options['hiddenLabel']) ? '' :  ' ' . "sr-only";
```
Comment (richard67) [inline layouts/joomla/form/renderfield.php]: @brianteeman What happens if someone adds a field where the `hiddenLabel` option is not empty but false: `hiddenLabel="false"`?

I think your code right now would also set the `sr-only` class in that case.

Comment (richard67) [reply layouts/joomla/form/renderfield.php]: Hmm, on the other hand I see it is like that without your PR, too.

Comment (brianteeman) [reply layouts/joomla/form/renderfield.php]: exactly ;)

Comment (richard67) [reply layouts/joomla/form/renderfield.php]: you wanna fix that, too? or wanna leave this pr as it is and i shall test?

**Comment by richard67** [comment]
I have tested this item :white_check_mark: successfully on f855283f459736696d40cc470d1d35b3a568182e<br /><br />I tested this with the TinyMCE editor plugin, which has such a field (field with name="configuration" in fieldset with name="basic").
Without this PR, the field is not rendered. With this PR, it is rendered with sr-only class.<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/25085">issues.joomla.org/tracker/joomla-cms/25085</a>.</sub>

File: layouts/joomla/form/renderfield.php (hunk 1 above)
Comment (Quy) [inline layouts/joomla/form/renderfield.php]:
```suggestion
$hide  = empty($options['hiddenLabel']) ? '' :  ' sr-only';
```

**Comment by richard67** [comment]
I have tested this item :white_check_mark: successfully on de6db064282162890c2f8c7e7955df39aa8610bd<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/25085">issues.joomla.org/tracker/joomla-cms/25085</a>.</sub>

**Comment by Quy** [comment]
I have tested this item :white_check_mark: successfully on de6db064282162890c2f8c7e7955df39aa8610bd<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/25085">issues.joomla.org/tracker/joomla-cms/25085</a>.</sub>

**Comment by Quy** [comment]
RTC<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/25085">issues.joomla.org/tracker/joomla-cms/25085</a>.</sub>

File: layouts/joomla/form/renderfield.php [hunk 2]
```diff
@@ -30,12 +30,11 @@
 $class = empty($options['class']) ? '' : ' ' . $options['class'];
 $rel   = empty($options['rel']) ? '' : ' ' . $options['rel'];
 $id    = $name . '-desc';
+$hide  = empty($options['hiddenLabel']) ? '' :  ' sr-only';
```
Comment (Quy) [inline layouts/joomla/form/renderfield.php]:
```suggestion
$hide  = empty($options['hiddenLabel']) ? '' : ' sr-only';
```

Comment (Quy) [reply layouts/joomla/form/renderfield.php]: Sorry!

**Comment by richard67** [comment]
I have tested this item :white_check_mark: successfully on faa2a56a85129d29459b6b75e5964061575116e7<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/25085">issues.joomla.org/tracker/joomla-cms/25085</a>.</sub>

**Comment by wilsonge** [comment]
Added docs required because technically we have a b/c break here requiring the label on hiddenLabel fields (I think the chances are low anyone's doing that but anyhow...)

**Comment by brianteeman** [comment]
Thanks

**Pull request merged**